tests/api/core/strapi/document-service/validation/validation.test.api.ts

```diff
@@ -0,0 +1,242 @@
+import type { Core, Modules } from '@strapi/types';
+import { errors } from '@strapi/utils';
+import { createTestSetup, destroyTestSetup } from '../../../../utils/builder-helper';
+import resources from '../resources/index';
+import { ARTICLE_UID } from '../utils';
+
+let strapi: Core.Strapi;
+
+describe('Document Service Validations', () => {
+  let testUtils;
+
+  beforeAll(async () => {
+    testUtils = await createTestSetup(resources);
+    strapi = testUtils.strapi;
+  });
+
+  afterAll(async () => {
+    await destroyTestSetup(testUtils);
+  });
+
+  const methods = [
+    'findMany',
+    'findFirst',
+    'findOne',
+    'publish',
+    'delete',
+    'create',
+    'unpublish',
+    'clone',
+    'update',
+  ];
+
+  describe.each(methods)('%s method', (methodName) => {
+    describe('sort', () => {
+      it('should not throw on existing attribute name', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({ sort: 'title' });
+      });
+
+      it('should not throw on private attribute', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({ sort: 'private' });
+      });
+
+      it('should not throw on password attribute', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({ sort: 'password' });
+      });
+
+      it('should not throw on existing nested (object) key', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({
+          populate: { categories: { sort: { name: 'asc' } } },
+        });
+      });
+
+      it('should not throw on existing nested (dot separated) key', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({
+          sort: 'categories.name',
+          populate: 'categories',
+        });
+      });
+
+      it('should throw ValidationError on invalid key', async () => {
+        await expect(
+          strapi.documents(ARTICLE_UID)[methodName]({ sort: 'fakekey' })
+        ).rejects.toThrow(errors.ValidationError);
+      });
+    });
+
+    describe('filters', () => {
+      it('should not throw on existing attribute equality', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({
+          filters: {
+            title: 'Hello World',
+          },
+        });
+      });
+
+      it('should not throw on private attribute', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({
+          filters: {
+            private: 'Hello World',
+          },
+        });
+      });
+
+      it('should not throw on password attribute', async () => {
+        await strapi.documents(ARTICLE_UID)[methodName]({
+          filters: {
+            password: 'Hello World',
+          },
+        });
+      });
```

Comment:
This is outside the scope of filter validation I think and maybe I'm missing something but here if I do      

it('should not throw on password attribute', async () => {
    const res = await strapi.documents(ARTICLE_UID)[methodName]({
        filters: {
        password: 'Hello World',
        },
    });

    console.log(res);
});

I get this for `findMany`. Should we be ablke to filter on passwords like this?

```
    [
      {
        id: 1,
        documentId: 'Article1',
        title: 'Article1-Draft-EN',
        password: '$2a$10$4EP.Y8crWYRpfk2rVmmoce8raDqGYVwMlcI//GDQdO6z06bO50igu',
        private: 'private',
        createdAt: '2024-08-27T10:56:51.245Z',
        updatedAt: '2024-08-27T10:56:51.245Z',
        publishedAt: null,
        locale: 'en'
      },
      {
        id: 2,
        documentId: 'Article2',
        title: 'Article2-Draft-EN',
        password: null,
        private: 'private',
        createdAt: '2024-08-27T10:56:51.249Z',
        updatedAt: '2024-08-27T10:56:51.249Z',
        publishedAt: null,
        locale: 'en'
      }
    ]
```

Reply:
On the doc service, yes, since it's the programmatic part and it should allow things that the content API doesn't (passwords, private fields, etc), it's more like a helper for writing db queries.

If the content API allows passwords or private fields, that would be a problem (and should be double-checked here just to be sure I haven't somehow allowed it under some circumstances)

Reply:
Also, even if someone uses the doc service to retrieve this data and then returns it directly to a controller, the controller _should_ be sanitizing the response to the user which will remove things like passwords based on the user permissions.